ocs_ci/ocs/longevity.py

```diff
@@ -555,6 +558,255 @@ def stage_0(
+            if cloned_pvc_obj.get_pvc_vol_mode == constants.VOLUME_MODE_BLOCK:
+                cloned_pod_objs.append(
+                    pod_factory(
+                        pvc=cloned_pvc_obj,
+                        raw_block_pv=True,
+                        status=constants.STATUS_RUNNING,
+                        pod_dict_path=constants.CSI_RBD_RAW_BLOCK_POD_YAML,
+                    )
+                )
+            else:
+                cloned_pod_objs.append(
+                    pod_factory(pvc=cloned_pvc_obj, status=constants.STATUS_RUNNING)
+                )
+
+        # Verify that the md5sum matches
+        for pod_obj in cloned_pod_objs:
+            log.info(f"Verifying md5sum of {file_name} " f"on pod {pod_obj.name}")
+            is_block = (
+                True
+                if pod_obj.pvc.get_pvc_vol_mode == constants.VOLUME_MODE_BLOCK
+                else False
+            )
+            file_name_pod = (
+                file_name
+                if not is_block
+                else pod_obj.get_storage_path(storage_type="block")
+            )
+            pod.verify_data_integrity(pod_obj, file_name_pod, pod_obj.pvc.parent.md5sum)
+            log.info(
+                f"Verified: md5sum of {file_name} on pod {pod_obj.name} "
+                f"matches with the original md5sum"
+            )
+
+        # Create Snapshots
+        snapshots = multi_snapshot_factory(
+            pvc_obj=pvc_objs, snapshot_name_suffix=namespace
+        )
+        log.info("Created snapshots from all the PVCs and snapshots are in Ready state")
+
+        # Restore Snapshots
+        restored_pvc_objs = multi_snapshot_restore_factory(
+            snapshot_obj=snapshots, restore_pvc_suffix="restore"
+        )
+        log.info("Created new PVCs from all the snapshots")
+
+        # Expand original PVCs
+        for pvc_obj in pvc_objs:
+            log.info(f"Expanding size of PVC {pvc_obj.name} to {pvc_size_new}G")
+            pvc_obj.resize_pvc(pvc_size_new, True)
+
+        total_pvcs = pvc_objs + cloned_pvcs + restored_pvc_objs
+        total_pods = pod_objs + cloned_pod_objs
+
+        # PVC and PV Teardown
+        pv_objs = list()
+        for pvc_obj in total_pvcs:
+            teardown_factory(pvc_obj)
+            pv_objs.append(pvc_obj.backed_pv_obj.name)
+            teardown_factory(pvc_obj.backed_pv_obj)
+
+        # POD Teardown
+        for pod_obj in total_pods:
+            teardown_factory(pod_obj)
+
+        # Delete PODs
+        pod_delete = executor.submit(delete_pods, total_pods)
+        pod_delete.result()
+
+        log.info("Verified: Pods are deleted.")
+
+        # Delete PVCs
+        pvc_delete = executor.submit(delete_pvcs, total_pvcs)
+        res = pvc_delete.result()
+        if not res:
+            raise ex.UnexpectedBehaviour("Deletion of PVCs failed")
+        log.info("PVC deletion was successful.")
+
+        log.info(f"------------ENDING CYCLE:{cycle_no}------------")
+
```

Comment:
As discussed, lets also have concurrent PVC clone, snapshot, expand covered in stage4.

Reply:
Added it.